src/main/java/org/openstreetmap/atlas/geography/atlas/packed/PackedAtlasSerializer.java

```diff
@@ -85,24 +94,44 @@ protected static PackedAtlas load(final Resource resource)
         // Assign the serializer to the Atlas! Then the Atlas will load all the fields depending on
         // demand.
         serializer.assign();
-        // TODO This is for backwards compatibility and will slow Atlas loading
-        // Try loading the meta data to make sure the data format is appropriate.
-        try
-        {
-            atlas.metaData();
-        }
-        catch (final CoreException e)
-        {
-            throw new CoreException(META_DATA_ERROR_MESSAGE, e);
-        }
+
+        // This is for backwards compatibility and will slow Atlas loading
+        determineAtlasLoadFormat(atlas);
+
         return atlas;
     };
 
+    /*
+     * Try loading the meta data to make sure the data format is appropriate. Keep trying formats
+     * until we find the right one
+     */
+    private static void determineAtlasLoadFormat(final PackedAtlas atlas)
+    {
+        final AtlasSerializationFormat[] possibleFormats = AtlasSerializationFormat.values();
+        for (final AtlasSerializationFormat candidateFormat : possibleFormats)
+        {
+            logger.info("Trying load format {}", candidateFormat);
+            atlas.setLoadSerializationFormat(candidateFormat);
```

Comment:
I would keep that load format local to the `PackedAtlasSerializer` here as it is very dependent on the resource being read.